int64 cycle_num
int64 step_length
int64 side_length
float32 angle
---
bool success
